heron/common/src/cpp/metrics/metricsmgr-client.cpp

```diff
@@ -63,6 +63,12 @@ void MetricsMgrClient::HandleClose(NetworkErrorCode) {
 
 void MetricsMgrClient::ReConnect() { Start(); }
 
+void MetricsMgrClient::SetPublisherPort(const sp_int32 _port) {
+  if (0 == port_ && 0 != _port) {
```

Comment:
We saw this test failure a lot and we suffer from it. Do you have any ETA when your refactor could be merged?
How about I fix this test failure first and then later align to your changes when your refactoring is available? Will create an issue for tracking this